### PR TITLE
Added Compiler Design Course

### PR DESCRIPTION
**Pull request by KaranKulshrestha**
I added the course under compilers section which is of high quality and absolutely free available on youtube

## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -273,6 +273,7 @@
 ### Compilers
 
 * [Compilers](https://www.youtube.com/playlist?list=PL6KMWPQP_DM97Hh0PYNgJord-sANFTI3i) - Ghassan Shobaki (YouTube)
+* [Neso's Compilers](https://youtube.com/playlist?list=PLBlnK6fEyqRjT3oJxFXRgjPNzeS-LFY-q) - Neso Academy (YouTube)
```
Comment (eshellman) [inline courses/free-courses-en.md:276]: Don't invent titles! Use the title given by the resource.

Comment (davorpa) [reply courses/free-courses-en.md:276]: Fixing as suggested by Eric...

```suggestion
* [Compiler Design](https://youtube.com/playlist?list=PLBlnK6fEyqRjT3oJxFXRgjPNzeS-LFY-q) - Neso Academy
```

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -273,6 +273,7 @@
 ### Compilers
 
 * [Compilers](https://www.youtube.com/playlist?list=PL6KMWPQP_DM97Hh0PYNgJord-sANFTI3i) - Ghassan Shobaki (YouTube)
+* [Compiler Design](https://youtube.com/playlist?list=PLBlnK6fEyqRjT3oJxFXRgjPNzeS-LFY-q) - Neso Academy
```
Comment (davorpa) [inline courses/free-courses-en.md:276]: Alphabetize

```suggestion
* [Compiler Design](https://youtube.com/playlist?list=PLBlnK6fEyqRjT3oJxFXRgjPNzeS-LFY-q) - Neso Academy
* [Compilers](https://www.youtube.com/playlist?list=PL6KMWPQP_DM97Hh0PYNgJord-sANFTI3i) - Ghassan Shobaki (YouTube)
```

**Review by davorpa — APPROVED** [review]

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**